### PR TITLE
CVS-41744 Integrate pipeline buildResponse with GetModelMeta gRPC/REST endpoints

### PR DESCRIPTION
**Pull request by dkalinowski**
https://jira.devtools.intel.com/browse/CVS-41744

### CODE REVIEW

**Comment by dkalinowski** [comment]
This is final merge request to enable GetModelMetadata functionality for pipelines.

File: src/pipeline_factory.hpp [hunk 1]
```diff
@@ -56,6 +56,16 @@ class PipelineFactory {
         const tensorflow::serving::PredictRequest* request,
         tensorflow::serving::PredictResponse* response,
         ModelManager& manager) const;
+
+    PipelineDefinition* findDefinitionByName(const std::string& name) const {
```
Comment (atobiszei) [inline src/pipeline_factory.hpp:60]: Why not reference?

Comment (dkalinowski) [reply src/pipeline_factory.hpp:60]: What if there is no definition with such name?   We might call ::definitionExists(name) before but that would require accessing definition map twice.

**Review by mzegla — APPROVED** [review]
Looks good

**Review by atobiszei — APPROVED** [review]

**Pull request merged**